pkg/bootstrap/platform/gcp.go

```diff
@@ -290,19 +290,19 @@ func zoneToRegion(z string) (string, error) {
 // Locality returns the GCP-specific region and zone.
 func (e *gcpEnv) Locality() *core.Locality {
 	var l core.Locality
+	_, _, _, envLoc := parseGCPMetadata()
 	if e.shouldFillMetadata() {
-		z, zerr := metadata.Zone()
-		if zerr != nil {
-			log.Warnf("Error fetching GCP zone: %v", zerr)
+		if envLoc != "" {
+			log.Warnf("Error fetching GCP zone: %v", envLoc)
 			return &l
 		}
-		r, rerr := zoneToRegion(z)
+		r, rerr := zoneToRegion(envLoc)
```

Comment:
what if env var is not set? Now this function will never succeed

Reply:
thanks, I missed this case. Updated